components/NavBar.tsx

```diff
@@ -1,46 +1,55 @@
-// CSS
+import { useState } from 'react';
 import styles from '../styles/components/NavBar.module.scss';
-// Config
 import { page_list } from '../config';
-// Components
 import Button, { ButtonType } from '../components/Button';
+import Logo from '../components/Logo';
 
-// Component
 interface Props {
   currentPage: string;
 }
+
 export default function NavBar(props: Props) {
-  // Build The Nav
-  const navContent: JSX.Element[] = [];
-  page_list.forEach((page, index) => {
-    if (!page.display_in_navbar) return;
-    navContent.push(
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+
+  const navContent = page_list
+    .filter((page) => page.display_in_navbar)
+    .map((page, index) => (
       <li
         key={index}
-        className={
-          props.currentPage == page.page_name ? styles.currentPage : ''
-        }
+        className={`${props.currentPage === page.page_name ? styles.currentPage : ''} w-full md:w-auto`}
       >
-        <Button
-          type={ButtonType.NAVBAR}
-          href={page.page_link}
-          className={
-            props.currentPage == page.page_name
-              ? ButtonType.NAVBAR_ACTIVE
-              : ButtonType.NAVBAR
-          }
-          label={page.page_name}
-        />
+        {props.currentPage === page.page_name ? (
```

Comment:
Why did you refactor this from just changing the class we use? The class approach allows us to keep things more centralized and we are only changing the one property.